fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OptExpressionDuplicator.java

```diff
@@ -120,12 +124,22 @@ public OptExpression visitLogicalTableScan(OptExpression optExpression, Void con
             ImmutableMap<Column, ColumnRefOperator> newColumnMetaToColRefMap = columnMetaToColRefMapBuilder.build();
             scanBuilder.setColumnMetaToColRefMap(newColumnMetaToColRefMap);
 
+            // process HashDistributionSpec
+            if (optExpression.getOp() instanceof LogicalOlapScanOperator) {
+                LogicalOlapScanOperator olapScan = (LogicalOlapScanOperator) optExpression.getOp();
+                HashDistributionSpec newHashDistributionSpec =
+                        processHashDistributionSpec(olapScan.getDistributionSpec(), columnRefFactory, columnMapping);
+                LogicalOlapScanOperator.Builder olapScanBuilder = (LogicalOlapScanOperator.Builder) scanBuilder;
```

Comment:
what if it's not a OlapScan

Reply:
there is a check of optExpression.getOp() instanceof LogicalOlapScanOperator